Allow `RGBA8::new_scaled` to run in `const`

`nx::gpu::canvas::RGBA8::new_scaled` is just a convenience wrapper for creating a new `RGBA8` with specific colors.

Unfortunately, as it is not marked `const`, it can't be used to e.g. create constants:

```rs
const C_BLACK: RGBA8 = RGBA8::new_scaled(0, 0, 0, 255);
```

```
error[E0015]: cannot call non-const associated function `RGBA8::new_scaled` in constants
  --> src/main.rs:65:28
   |
65 |     const C_BLACK: RGBA8 = RGBA8::new_scaled(0, 0, 0, 255);
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
```

But there's nothing stopping the user from just manually

```rs
const C_BLACK: RGBA8 = RGBA8::new().with_r(0).with_g(0).with_b(0).with_a(255);
```


The simple fix is to just add the keyword `const` on `new_scaled`, both for `RGBA8` and `RGBA4`. This will allow users to use it either at compiletime or runtime.

--- a/src/gpu/canvas.rs
+++ b/src/gpu/canvas.rs
@@ -46,7 +46,7 @@ pub struct RGBA4 {
 impl RGBA4 {
     /// Takes in standard 8-bit colors and scales them down to fit the range of 4-bit channels (via 4-bit right shift)
     #[inline]
-    pub fn new_scaled(r: u8, g: u8, b: u8, a: u8) -> Self {
+    pub const fn new_scaled(r: u8, g: u8, b: u8, a: u8) -> Self {
         Self::new()
             .with_r(r >> 4)
             .with_g(g >> 4)
@@ -127,7 +127,7 @@ pub struct RGBA8 {
 }
 
 impl RGBA8 {
-    pub fn new_scaled(r: u8, g: u8, b: u8, a: u8) -> Self {
+    pub const fn new_scaled(r: u8, g: u8, b: u8, a: u8) -> Self {
         Self::new().with_r(r).with_g(g).with_b(b).with_a(a)
     }
 

